BlockchainSdk/Common/Interfaces/TransactionValidator/FeeResourceRestrictable.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  FeeResourceRestrictable.swift
+//  BlockchainSdk
+//
+//  Created by Aleksei Muraveinik on 06.06.24.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+
+protocol FeeResourceRestrictable {
+    func validateFeeResource(amount: Amount, fee: Amount) async throws
+}
+
+extension FeeResourceRestrictable where Self: WalletProvider {
+    func validateFeeResource(amount: Amount, fee: Amount) throws {
+        let currentFeeResource = wallet.amounts[fee.type]?.value ?? .zero
+        let maxFeeResource = wallet.amounts[amount.type]?.value ?? .zero
```

Comment:
По коду это получается просто баланс того что отправляем ?
Лучше бы тогда так)
```suggestion
        guard let balance = wallet.amounts[amount.type]?.value else {
            throw ValidationError.balanceNotFound
        }
```

P.S: если это так принято называть в сети, то можно и так оставить, пофиг

Reply:
Мы отправляем KOIN, в качестве комиссии используем MANA

currentFeeResource - MANA
maxFeeResource - MANA, но это число равно количеству KOIN в кошельке

Это можно было бы назвать `balance`, но семантически это используется именно в качестве максимального количества `feeResource`, так что я бы оставил нейминг